packages/core/src/codecs/octetstream-codec.ts

```diff
@@ -542,6 +543,7 @@ export default class OctetstreamCodec implements ContentCodec {
             throw new Error("Missing 'length' parameter necessary for write");
         }
 
+        const offset = schema["ex:bitOffset"] !== undefined ? parseInt(schema["ex:bitOffset"]) : 0;
```

Comment:
If you want to introduce the checks it would be fine by me.. 👍

Reply:
I added some more checks (and tests making sure the checks work) in my last commit

Reply:
A "simpler" approach could have been a helper method that does the check (since I think the are almost all the same). Anyhow, fine like this as well 👍